test/integration/compiler.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { Expect, Test, TestCase } from "alsatian";
+import { execSync } from "child_process"
+import { existsSync, removeSync, unlink } from "fs-extra"
+import { resolve } from "path"
+
+export class CompilerTests {
+
+    @TestCase('tsconfig.default.json', ['typescript_lualib.lua', 'nested/folder/file.lua'])
+    @TestCase('tsconfig.outDir.json', ['outDir/typescript_lualib.lua', 'outDir/nested/folder/file.lua'])
+    @TestCase('tsconfig.rootDir.json', ['nested/typescript_lualib.lua', 'nested/folder/file.lua'])
+    @TestCase('tsconfig.bothDirOptions.json', ['outDir/typescript_lualib.lua', 'outDir/folder/file.lua'])
+    @Test("Compile project")
+    public compileProject(tsconfig: string, expectedFiles: string[]) {
+        const compilerPath = resolve('dist/Compiler.js');
+        const tsconfigPath = resolve('test/integration/project', tsconfig);
+
+        // Compile project
+        execSync(`node ${compilerPath} -p ${tsconfigPath}`);
+
+        expectedFiles.forEach(relativePath => {
+            const absolutePath = resolve('test/integration/project', relativePath);
+
+            // Assert
+            Expect(existsSync(absolutePath)).toBe(true);
+
+            // Delete file
+            unlink(absolutePath, (error) => {
+                throw error;
+            });
+        });
+
+        // Delete outDir folder
+        removeSync('test/integration/project/outDir');
+    }
+
```

Comment:
I am not sure this is a good way to test the compiler.

Reply:
These tests are about `outDir` and `rootDir` and I think those are appropriately tested with that. But I'd be happy to implement something else if you tell me what you think it should be instead.

Reply:
Right makes sense. Might want to rename the test.
I was thinking about checking if paths are resolved correctly directly, but this is hard to test since all the resolve stuff is packed into parseCommandLine() and compile() which makes it hard to test directly. So checking if a file exists is probably the best way right now. I might revisit this when I rewrite the CLI interface.